### PR TITLE
Show (*) and a tooltip on automatically closed staff attendances

### PR DESCRIPTION
**Pull request by reynders**
Screenshot 2024-01-29 at 12 33 17

### CODE REVIEW

File: frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx [hunk 1]
```diff
@@ -478,6 +478,16 @@ const AttendanceRow = React.memo(function AttendanceRow({
   const { i18n } = useTranslation()
   const today = LocalDate.todayInHelsinkiTz()
 
+  const attendanceTooltipText = (attendance: Attendance) => (
+    <div>
+      {attendance.departedAutomatically ? (
```
Comment (reynders) [inline frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx:483]: Redundant check for future use of this tooltip for other info too

File: frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx [hunk 2]
```diff
@@ -564,14 +574,33 @@ const AttendanceRow = React.memo(function AttendanceRow({
...
+                          <AttendanceCell>
+                            <AttendanceTime data-qa="arrival-time">
+                              {renderTime(attendance.arrived, date)}
+                            </AttendanceTime>
+                            <AttendanceTime data-qa="departure-time">
+                              {renderTime(attendance.departed, date)}
+                            </AttendanceTime>
+                          </AttendanceCell>
+                        )}
+                      </>
```
Comment (Joosakur) [inline frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx:603]: I think duplicating the whole thing is unnecessary. AFAIK tooltip is not rendered if tooltip prop is falsy.

```
const attendanceTooltip = attendanceTooltipText(attendance) // undefined if no tooltip
...
                          <Tooltip
                            key={i}
                            tooltip={attendanceTooltip}
                            data-qa="attendance-tooltip"
                          >
                            <AttendanceCell>
                              <AttendanceTime data-qa="arrival-time">
                                {renderTime(attendance.arrived, date)}
                              </AttendanceTime>
                              <AttendanceTime data-qa="departure-time">
                                {renderTime(attendance.departed, date)}
                                {attendanceTooltip ? ' (*)'}
                              </AttendanceTime>
                            </AttendanceCell>
                          </Tooltip>
```

File: frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx [hunk 3]
```diff
@@ -478,6 +478,16 @@ const AttendanceRow = React.memo(function AttendanceRow({
   const { i18n } = useTranslation()
   const today = LocalDate.todayInHelsinkiTz()
 
+  const attendanceTooltipText = (attendance: Attendance) => (
+    <div>
+      {attendance.departedAutomatically ? (
+        <div>{i18n.unit.staffAttendance.departedAutomatically}</div>
+      ) : (
+        <div>{i18n.unit.staffAttendance.departedManually}</div>
```
Comment (Joosakur) [inline frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx:486]: Is departedManually ever shown? Below the tooltip is omitted when `!attendance.departedAutomatically`. You can just return undefined here.

File: frontend/src/lib-customizations/defaults/employee/i18n/fi.tsx [hunk 4]
```diff
@@ -2518,7 +2518,9 @@ export const fi = {
         name: 'Nimi',
         namePlaceholder: 'Sukunimi Etunimi',
         group: 'Ryhmä'
-      }
+      },
+      departedManually: 'Manuaalikatkaistu',
```
Comment (Joosakur) [inline frontend/src/lib-customizations/defaults/employee/i18n/fi.tsx:2522]: probably not needed

**Review by Joosakur — CHANGES_REQUESTED** [review]

File: frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx [hunk 5]
```diff
@@ -564,14 +571,21 @@ const AttendanceRow = React.memo(function AttendanceRow({
...
+                          <AttendanceTime data-qa="arrival-time">
+                            {renderTime(attendance.arrived, date)}
+                          </AttendanceTime>
+                          <AttendanceTime data-qa="departure-time">
+                            {renderTime(attendance.departed, date)}
+                            {attendance.departedAutomatically ? `(*)` : ''}
+                          </AttendanceTime>
+                        </AttendanceCell>
+                      </Tooltip>
                     ))
```
Comment (Joosakur) [inline frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx:589]:
```
                    attendancesForToday.map((attendance, i) => {
                      const tooltip = attendanceTooltipText(attendance)
                      return (
                        <Tooltip
                          key={i}
                          tooltip={tooltip}
                          data-qa="attendance-tooltip"
                        >
                          <AttendanceCell>
                            <AttendanceTime data-qa="arrival-time">
                              {renderTime(attendance.arrived, date)}
                            </AttendanceTime>
                            <AttendanceTime data-qa="departure-time">
                              {renderTime(attendance.departed, date)}
                              {tooltip && ' (*)'}
                            </AttendanceTime>
                          </AttendanceCell>
                        </Tooltip>
                      )
                    })
```
This way you aren't duplicating logic and adding another tooltip reason only requires changing the tooltip returning function.

**Review by Joosakur — APPROVED** [review]

**Pull request merged**